File: Interpreted/lev-distance-recursive/lev.py
import sys

def leven(s, t):

    if s == '': return len(t)
    if t == '': return len(s)

    s1 = s[1:]
    t1 = t[1:]

    if s[0] == t[0]:
        return leven(s1, t1)

    return 1 + min(
        leven(s,  t1),
        leven(s1, t ),
        leven(s1, t1)
    )

print(leven(sys.argv[1], sys.argv[2]))

File: Interpreted/lev-distance-recursive/lev.php
<?php
function leven($s, $t) {

    if ($s == '') {
        return strlen($t);
    }
    if ($t == '') {
        return strlen($s);
    }

    $s1 = substr($s, 1);
    $t1 = substr($t, 1);

    return ($s[0] == $t[0]
        ? leven($s1, $t1)
        : 1 + min(
                leven($s1, $t1),
                leven($s,  $t1),
                leven($s1,  $t)
              )
          );
}

echo leven($argv[1], $argv[2]), "\n";
?>
